tool/tsh/db.go

```diff
@@ -518,77 +555,171 @@ func getConnectCommand(cf *CLIConf, tc *client.TeleportClient, profile *client.P
+		if c.options.localProxyHost == "localhost" {
 			args = append(args, "--protocol", "TCP")
 		}
 	}
 
+	return args
+}
+
+func (c *cliCommandBuilder) getMariadbArgs() []string {
+	args := c.getMySQLCommonCmdOpts()
+	sslCertPath := c.profile.DatabaseCertPathForCluster(c.tc.SiteName, c.db.ServiceName)
+
+	args = append(args, []string{"--ssl-key", c.profile.KeyPath()}...)
+	args = append(args, []string{"--ssl-ca", c.profile.CACertPath()}...)
+	args = append(args, []string{"--ssl-cert", sslCertPath}...)
+
+	// Flag below verifies "Common Name" check on the certificate provided by the server.
+	if !c.tc.InsecureSkipVerify {
+		args = append(args, "--ssl-verify-server-cert")
+	}
+
+	return args
+}
+
+func (c *cliCommandBuilder) getMySQLOracleCommand() *exec.Cmd {
+	args := c.getMySQLCommonCmdOpts()
+
+	// defaults-group-suffix must be first.
+	groupSuffix := []string{fmt.Sprintf("--defaults-group-suffix=_%v-%v", c.tc.SiteName, c.db.ServiceName)}
+	args = append(groupSuffix, args...)
+
+	// override the ssl-mode from a config file is --insecure flag is provided to 'tsh db connect'.
+	if c.tc.InsecureSkipVerify {
+		args = append(args, fmt.Sprintf("--ssl-mode=%s", mysql.MySQLSSLModeVerifyCA))
+	}
+
 	return exec.Command(mysqlBin, args...)
 }
 
-func getMongoCommand(tc *client.TeleportClient, profile *client.ProfileStatus, db *tlsca.RouteToDatabase, host string, port int, options connectionCommandOpts) *exec.Cmd {
+// getMySQLCommand returns mariadb command if the binary is on the path. Otherwise,
+// mysql command is returned. Both mysql versions (MariaDB and Oracle) are supported.
+func (c *cliCommandBuilder) getMySQLCommand() *exec.Cmd {
+	// Check if mariadb client is available. Prefer it over mysql client even if connecting to MySQL server.
+	if c.isMariadbBinAvailable() {
+		args := c.getMariadbArgs()
+		return exec.Command(mariadbBin, args...)
+	}
+
+	// Check which flavor is installed. Otherwise, we don't know which ssl flag to use.
+	// Yes, at the moment of writing they both have the same name and accept different parameters.
+	mySQLMariadbFlavor, err := c.isMySQLBinMariaDBFlavor()
+	if mySQLMariadbFlavor && err == nil {
+		args := c.getMariadbArgs()
+		return exec.Command(mysqlBin, args...)
+	}
+
+	// Either we failed to check the flavor or binary comes from Oracle. Regardless return mysql command.
+	// This will generate "not found mysql" command.
+	return c.getMySQLOracleCommand()
+}
+
+// isMariadbBinAvailable returns true if "mariadb" binary is found in the system PATH.
+func (c *cliCommandBuilder) isMariadbBinAvailable() bool {
+	_, err := c.exe.LookPath(mariadbBin)
+	return err == nil
+}
+
+// isMySQLBinMariaDBFlavor checks if mysql binary comes from Oracle or MariaDB.
+// true is returned when binary comes from MariaDB, false when from Oracle.
+func (c *cliCommandBuilder) isMySQLBinMariaDBFlavor() (bool, error) {
+	// Check if mysql comes from Oracle or MariaDB
+	mysqlVer, err := c.exe.RunCommand(mysqlBin, "--version")
```

Comment:
Wouldn't it be a bit more graceful to check if `mysql` binary exists first and return an appropriate error, like with `mariadb`?

Reply:
Added